crates/hir_ty/src/inference.rs

```diff
@@ -214,7 +214,11 @@ impl<'a> InferBody<'a> {
 
     fn infer_stmt(&mut self, stmt: &hir::Stmt) -> bool {
         let ty = match stmt {
-            hir::Stmt::VariableDef { name: _, value } => {
+            hir::Stmt::VariableDef {
+                name: _,
+                is_mutability: _,
+                value,
+            } => {
                 let ty = self.infer_expr(*value);
                 let ty_scheme = TypeScheme::new(ty.clone());
                 self.mut_current_scope().insert(*value, ty_scheme);
```

Comment:
**Optimization Issue**: The method 'infer_expr' is called for each variable definition to determine the type, which might be inefficient if 'infer_expr' is computationally expensive or called frequently in a loop. This can lead to performance degradation, especially in large codebases or with complex nested structures. <br> **Fix**: Consider caching the results of 'infer_expr' for repeated calls with the same arguments within the same scope to avoid redundant computations. Implementing memoization or a similar caching mechanism could significantly reduce the computational overhead. <br> **Code Suggestion**: 
 ```
 + let ty = self.cached_infer_expr(*value);
 - let ty = self.infer_expr(*value);
 ```